account_payment_promissory_note/models/account_payment.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2018 Sergio Teruel <sergio.teruel@tecnativa.com>
+# Copyright 2018 Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountAbstractPayment(models.AbstractModel):
+    _inherit = 'account.abstract.payment'
+
+    deferred = fields.Boolean()
+    date_due = fields.Date(
+        string='Due Date',
+    )
+
+    @api.onchange('deferred')
+    def _onchange_deferred(self):
+        if not self.deferred:
+            self.date_due = False
+            return
+        invoices = False
+        if self._name == 'account.payment':
+            invoices = self.invoice_ids
+        elif self._name == 'account.register.payments':
+            active_ids = self._context.get('active_ids')
+            invoices = self.env['account.invoice'].browse(active_ids)
+        if invoices:
+            self.date_due = max(invoices.mapped('date_due'))
```

Comment:
Mention in the USAGE section about all the possible flows that I'm seeing here and also that the initial proposed due date is computed according this criteria.